test/support/test_helpers.ex

```diff
@@ -1,9 +1,25 @@
 defmodule CodeCorps.TestHelpers do
   use Phoenix.ConnTest
+  import ExUnit.Assertions
 
   def ids_from_response(response) do
     Enum.map response["data"], fn(attributes) ->
       String.to_integer(attributes["id"])
     end
   end
+
+  def assert_jsonapi_relationship(json = %{"relationships" => relationships}, relationship_name, id) do
+    assert relationships[relationship_name]["data"]["id"] == Integer.to_string(id)
+    json
+  end
+
+  def assert_jsonapi_relationship(json, relationship_name, id) do
+    assert json["data"]["relationships"][relationship_name]["data"]["id"] == Integer.to_string(id)
```

Comment:
Yes, this should fail. Since we aren't pattern matching having `assert_jsonapi_relationships(_, _, _)` would be unreachable.  So, this would throw an error if there were no `data` attribute in the `json` list passed.